nixos/modules/services/networking/ucarp.nix

```diff
@@ -0,0 +1,206 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.networking.ucarp;
+
+  ucarpExec = concatStringsSep " " (
+    [
+      "${cfg.package}/bin/ucarp"
+      "--interface=${cfg.interface}"
+      "--srcip=${cfg.srcIp}"
+      "--vhid=${toString cfg.vhId}"
+      "--addr=${cfg.addr}"
+      "--advbase=${toString cfg.advBase}"
+      "--advskew=${toString cfg.advSkew}"
+      "--upscript=${cfg.upscript}"
+      "--downscript=${cfg.downscript}"
+      "--deadratio=${toString cfg.deadratio}"
+    ]
+    ++ (optional (cfg.pass != null) "--pass=${cfg.pass}")
+    ++ (optional (cfg.passwordFile != null) "--passfile=${cfg.passwordFile}")
+    ++ (optional cfg.preempt "--preempt")
+    ++ (optional cfg.neutral "--neutral")
+    ++ (optional cfg.shutdown "--shutdown")
+    ++ (optional cfg.ignoreIfState "--ignoreifstate")
+    ++ (optional cfg.noMcast "--nomcast")
+    ++ (optional (cfg.extraParam != null) "--xparam=${cfg.extraParam}")
+  );
+in {
+  options.networking.ucarp = {
+    enable = mkEnableOption "ucarp, userspace implementation of CARP";
+
+    interface = mkOption {
+      type = types.str;
+      description = "Network interface to bind to.";
+      example = "eth0";
+    };
+
+    srcIp = mkOption {
+      type = types.str;
+      description = "Source (real) IP address of this host.";
+    };
+
+    vhId = mkOption {
+      type = types.ints.between 1 255;
+      description = "Virtual IP identifier shared between CARP hosts.";
+      example = 1;
+    };
+
+    pass = mkOption {
+      type = types.nullOr types.str;
+      description = ''
+        Shared password between CARP hosts.
+
+        Either <literal>pass</literal> or <literal>passwordFile</literal> must
+        be set.
+      '';
+      default = null;
+    };
+
+    passwordFile = mkOption {
```

Comment:
I suggest we do not make it easy for people. If someone really wants to shoot themselves in the foot, or has a legitimate use case of why it is OK to store a password in the nix store, they can always do this:
```
networking.ucarp.passwordFile = pkgs.writeText "badidea.txt" "password-everyone-on-this-server-can-read";
```
```